frontend-react/src/hooks/UseOktaMemberships.ts

```diff
@@ -169,20 +177,24 @@ const calculateNewState = (
             };
             storeOrganizationOverride(JSON.stringify(newActive));
             return newState;
+        case MembershipActionType.INITIALIZE:
+            return { ...state, initialized: true };
         case MembershipActionType.RESET:
-            return defaultState;
+            return { ...defaultState, initialized: state.initialized };
         default:
             return state;
     }
 };
 
 // try to read from existing stored state
+// since this is only called on first render, initialized will always be `false`
+// even though we are (needlessly) storing and receving initialized values from store
 export const getInitialState = () => {
     const storedState = getSessionMembershipState();
     const storedStateWithOverride = calculateMembershipsWithOverride(
         storedState || {}
     );
-    return { ...defaultState, ...storedStateWithOverride };
+    return { ...defaultState, ...storedStateWithOverride, initialized: false };
```

Comment:
This could be problematic. I'd advocate for `initialized: false` to be moved into `defaultState` so storedStateWithOverride's initialized value is present on override.

Reply:
yeah - as discussed this is to make sure that the `false` from the defaultState doesn't get overwritten by whatever is stored in state. We always want to make sure we initialize with a `false` value here, and let the hook do the work on second render